main.py

```diff
@@ -165,36 +248,53 @@ async def submit_experiment(  # pylint: disable=too-many-arguments
           None for no due date.
         visualize: If True, the experiment file is modified for visualization.
           The original file and vcd file are saved in the visualize path set in config.json.
+        cls: The class name of the experiment.
+          It is only necessary when the visualize argument is True.
     
     Returns:
         The run identifier, an integer which is incremented at each experiment submission.
     """
     if visualize:
+        submission_time = datetime.now().isoformat(timespec="seconds")
         experiment_path = posixpath.join(configs["master_path"], configs["repository_path"], file)
+        visualize_dir_path = posixpath.join(configs["master_path"], configs["visualize_path"])
+        modified_experiment_path = posixpath.join(
+            visualize_dir_path,
+            f"experiment_{submission_time}.py"
+        )
+        # modify the experiment code
         with open(experiment_path, encoding="utf-8") as experiment_file:
-            _code = experiment_file.read()
-        # TODO(BECATRUE): The code will be modifed in #37.
+            code = experiment_file.read()
+        modified_code = modify_experiment_code(code, cls)
+        # save the modified experiment code
+        with open(modified_experiment_path, "w", encoding="utf-8") as modified_experiment_file:
+            modified_experiment_file.write(modified_code)
+        submission_file_path = modified_experiment_path
     else:
-        # TODO(BECATRUE): The exact experiment path will be assigned in #37.
-        pass
+        submission_file_path = posixpath.join(configs["repository_path"], file)
+    args_dict = json.loads(args)
     expid = {
```

Comment:
What was `expid`? Please rename it with full words.

Reply:
As far as I know, this `expid` came from the original artiq server.

Reply:
Yes! As Giwon said, `expid` came from [`artiq_client.py`](https://github.com/m-labs/artiq/blob/master/artiq/frontend/artiq_client.py#L148).
Though it is an abbreviation and looks little vague, I think using original expression is better.